AssertionError hit in test_chained_logging_hits_log_timeout

<!-- READ THIS FIRST:
  - If you need additional help with this template, please refer to https://www.home-assistant.io/help/reporting_issues/
  - Make sure you are running the latest version of Home Assistant before reporting an issue: https://github.com/home-assistant/core/releases
  - Do not report issues for integrations if you are using custom components or integrations.
  - Provide as many details as possible. Paste logs, configuration samples and code into the backticks.
  DO NOT DELETE ANY TEXT from this template! Otherwise, your issue may be closed without comment.
-->
## The problem
<!-- 
  Describe the issue you are experiencing here to communicate to the
  maintainers. Tell us what you were trying to do and what happened.
-->

The test case `test_chained_logging_hits_log_timeout` is running into an AssertionError on one of my machines. It seems it is flaky.

It was introduced here https://github.com/home-assistant/core/pull/38311.


## Environment
<!--
  Provide details about the versions you are using, which helps us to reproduce
  and find the issue quicker. Version information is found in the
  Home Assistant frontend: Developer tools -> Info.
-->

- Home Assistant Core release with the issue: 0.113.3
- Last working Home Assistant Core release (if known): 0.103.0
- Operating environment (OS/Container/Supervised/Core): NixOS


## Traceback/Error logs
<!--
  If you come across any trace or error logs, please provide them.
-->

```txt
running install tests
============================= test session starts ==============================
platform linux -- Python 3.8.3, pytest-5.4.3, py-1.8.1, pluggy-0.13.1
rootdir: /build/source, inifile: setup.cfg, testpaths: tests
plugins: aiohttp-0.3.0, requests-mock-1.8.0
collected 1181 items / 14 deselected / 1167 selected

[...]

=================================== FAILURES ===================================
________________ test_chained_logging_hits_log_timeout[pyloop] _________________

hass = <homeassistant.core.HomeAssistant object at 0x7ffff32adca0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x7ffff32ad0a0>

    async def test_chained_logging_hits_log_timeout(hass, caplog):
        """Ensure we log which task is blocking startup when there is a task chain and debug logging is on."""
        caplog.set_level(logging.DEBUG)

        created = 0

        async def _task_chain_1():
            nonlocal created
            created += 1
            if created > 10:
                return
            hass.async_create_task(_task_chain_2())

        async def _task_chain_2():
            nonlocal created
            created += 1
            if created > 10:
                return
            hass.async_create_task(_task_chain_1())

        with patch.object(ha, "BLOCK_LOG_TIMEOUT", 0.1):
            hass.async_create_task(_task_chain_1())
            await hass.async_block_till_done()

>       assert "_task_chain_" in caplog.text
E       AssertionError: assert '_task_chain_' in ''
E        +  where '' = <_pytest.logging.LogCaptureFixture object at 0x7ffff32ad0a0>.text

tests/test_core.py:1443: AssertionError

```

--- a/tests/test_core.py
+++ b/tests/test_core.py
@@ -1436,14 +1436,14 @@ async def test_chained_logging_hits_log_timeout(hass, caplog):
     async def _task_chain_1():
         nonlocal created
         created += 1
-        if created > 10:
+        if created > 1000:
             return
         hass.async_create_task(_task_chain_2())
 
     async def _task_chain_2():
         nonlocal created
         created += 1
-        if created > 10:
+        if created > 1000:
             return
         hass.async_create_task(_task_chain_1())
 

